docs/OV_Runtime_UG/ov_infer_request.md

```diff
@@ -0,0 +1,255 @@
+
+`ov::InferRequest` allows to get input/output tensors by tensor name, index, port and without any arguments in case if model has only one input or output.
+
+  * `ov::InferRequest::get_input_tensor()`, `ov::InferRequest::set_input_tensor()`, `ov::InferRequest::get_output_tensor()`, `ov::InferRequest::set_output_tensor()` methods without arguments can be used to get or set input/output tensor for model with only one input/output:
+    @sphinxdirective
+
+    .. tab:: C++
+
+        .. doxygensnippet:: docs/snippets/ov_infer_request.cpp
+           :language: cpp
+           :fragment: [get_set_one_tensor]
+
+    .. tab:: Python
+
+        .. doxygensnippet:: docs/snippets/ov_infer_request.py
+           :language: python
+           :fragment: [get_set_one_tensor]
+
+    @endsphinxdirective
+
+  * `ov::InferRequest::get_input_tensor()`, `ov::InferRequest::set_input_tensor()`, `ov::InferRequest::get_output_tensor()`, `ov::InferRequest::set_output_tensor()` methods with argument can be used to get or set input/output tensor by input/output index:
+    @sphinxdirective
+
+    .. tab:: C++
+
+        .. doxygensnippet:: docs/snippets/ov_infer_request.cpp
+           :language: cpp
+           :fragment: [get_set_index_tensor]
+
+    .. tab:: Python
+
+        .. doxygensnippet:: docs/snippets/ov_infer_request.py
+           :language: python
+           :fragment: [get_set_index_tensor]
+
+    @endsphinxdirective
+
+  * `ov::InferRequest::get_tensor()`, `ov::InferRequest::set_tensor()` methods can be used to get or set input/output tensor by tensor name:
+    @sphinxdirective
+
+    .. tab:: C++
+
+        .. doxygensnippet:: docs/snippets/ov_infer_request.cpp
+           :language: cpp
+           :fragment: [get_set_tensor]
+
+    .. tab:: Python
+
+        .. doxygensnippet:: docs/snippets/ov_infer_request.py
+           :language: python
+           :fragment: [get_set_tensor]
+
+    @endsphinxdirective
+
+  * `ov::InferRequest::get_tensor()`, `ov::InferRequest::set_tensor()` methods can be used to get or set input/output tensor by port:
+    @sphinxdirective
+
+    .. tab:: C++
+
+        .. doxygensnippet:: docs/snippets/ov_infer_request.cpp
+           :language: cpp
+           :fragment: [get_set_tensor_by_port]
+
+    .. tab:: Python
+
+        .. doxygensnippet:: docs/snippets/ov_infer_request.py
+           :language: python
+           :fragment: [get_set_tensor_by_port]
+
+    @endsphinxdirective
+
+## Examples of InferRequest usages
+
+### Cascade of models
+
+`ov::InferRequest` can be used to organize cascade of models. You need to have infer requests for each model.
+In this case you can get output tensor from the first request using `ov::InferRequest::get_tensor()` and set it as input for the second request using `ov::InferRequest::set_tensor()`. But be careful, shared tensors across compiled models can be rewritten by the first model if the first infer request is run once again, while the second model has not started yet.
```

Comment:
can we also provide example with remote tensors? For devices which supports `ov::Core::get_default_context` ?